src/newrelic.coffee

```diff
@@ -34,16 +35,24 @@
 
 gist = require 'quick-gist'
 moment = require 'moment'
+diff = require 'fast-array-diff'
 
 plugin = (robot) ->
   apiKey = process.env.HUBOT_NEWRELIC_API_KEY
   apiHost = process.env.HUBOT_NEWRELIC_API_HOST or 'api.newrelic.com'
+  room = process.env.HUBOT_NEWRELIC_ALERT_ROOM
+  throw new Error('HUBOT_NEWRELIC_API_KEY is not set') unless apiKey
+  throw new Error('HUBOT_NEWRELIC_ALERT_ROOM is not set') unless room
+
   apiBaseUrl = "https://#{apiHost}/v2/"
+  maxMessageLength = 4000 # some chat servers have a limit of 4000 chars per message. Lame.
   config = {}
 
   config.up = ':white_check_mark:'
   config.down = ':no_entry_sign:'
-  config.blockquote = '>>> '
+
+  robot.brain.data.newrelicdata ?= {}
+  robot.brain.data.newrelicdata.violations ?= []
```

Comment:
I recommend using the official getters and setters for [brain persistence](https://github.com/github/hubot/blob/master/docs/scripting.md#persistence). e.g. `newrelicdata = robot.brain.get 'newrelicdata'`

Here's a kicker, though: It's a bad idea to read or write to the brain this early on in a script. Hubot consists of several event emitters (see https://github.com/github/hubot/issues/542#issuecomment-21461435) that fire events in an unreliable order and there's no guarantee that this script will be loaded before the brain is loaded. I've found it's best to wait to read from the brain until you actually need the data (i.e. line 102).